ext/drreg/drreg.h

```diff
@@ -459,6 +459,11 @@ DR_EXPORT
  * \p *swap.  It is up to the caller to un-reserve the register in
  * that case.
  *
+ * For regs that were reserved in a prior phase also, this routine
+ * restores the reg's meta value from that prior phase, OR if the
```

Comment:
This doesn't seem good enough: users don't know what registers the scatter/gather expansion used, and so users can't write correct clients.  It seems like we have to not overlap scratch registers between app2app and insertion or something, or have the scatter expansion provide guarantees on its behavior so a client knows where it can get app values.

A real-world case to focus on is #3995 where one method of implementation is phases of instrumentation implemented with flushes.  With flushes invoked from clean calls, the call has to get the proper app state.  Maybe having just one point at the last instruction of the basic block where the state is available is good enough there.